### PR TITLE
Fix override option for validate_regex

### PR DESCRIPTION
**Pull request by dbernheisel**
# Problem
When supplying `validate_regex: false` on the input itself, it seems to be ignored

# Solution
Looks like I made a logic error when checking if the option is enabled. When the option is `false`, it failed the condition and then moved on to check the application config, which defaults to `true` when not present.

The change is to explicitly check for `false` in the options

### CODE REVIEW

File: lib/formulator/input.ex [hunk 1]
```diff
@@ -43,10 +43,12 @@ defmodule Formulator.Input do
...
   defp option_enabled?(options, field, default) do
-    Enum.any?([
-      options[field] == true,
-      Application.get_env(:formulator, field, default) == true,
-    ])
+    cond do
+      Keyword.get(options, field) == false -> false
+      Keyword.get(options, field) == true -> true
+      Application.get_env(:formulator, field, default) == true -> true
+      true -> false
```
Comment (drapergeek) [inline lib/formulator/input.ex]: Minor: what do you think about using `_` here instead of `true`. With all the lovely boolean logic here, that last part was a bit hard to follow for me.

```suggestion
      _ -> false
```
Do or do not, the choice is yours.

File: lib/formulator/input.ex [hunk 2]
```diff
@@ -43,10 +43,12 @@ defmodule Formulator.Input do
   defp add_format_validation_attribute(options, _, _), do: options
 
   defp option_enabled?(options, field, default) do
```
Comment (drapergeek) [inline lib/formulator/input.ex]: Given the logic here, do you think it's worth exposing this publicly and unit testing it?

File: lib/formulator/input.ex (hunk 1 above)
Comment (dbernheisel) [reply lib/formulator/input.ex]: Ya, sounds good.

File: lib/formulator/input.ex (hunk 2 above)
Comment (dbernheisel) [reply lib/formulator/input.ex]: I think having the tests on the behavior of the regex options is enough. That said, I think a couple more tests would be good. I'll add some more.

**Comment by drapergeek** [comment]
:shipit:
